src/utils/decodedSourcemap.ts

```diff
@@ -22,7 +22,27 @@ export function decodedSourcemap(map: Input): ExistingDecodedSourceMap | null {
 		};
 	}
 
-	const mappings = typeof map.mappings === 'string' ? decode(map.mappings) : map.mappings;
+	const decodedMap = {
+		...(map as ExistingRawSourceMap | ExistingDecodedSourceMap),
+		mappings: undefined as unknown as SourceMapMappings
```

Comment:
This is a little gross, but I wanted to exclude the string mappings from the input object, and `undefined` seemed like a better option than:
```ts
mappings: [] as SourceMapMappings
```
since that empty array would never be accessible.

If I just left it as `undefined` without the type, then I'd need to do a similar type assertion on the return statement.

Any preferences here?

Reply:
Ah yes... I always forget that accessors in object literals are a thing in JS an not just a feature from classes. I agree this is a much better solution.